src/unitxt/operator.py

```diff
@@ -176,16 +177,17 @@ def _process_single_stream(self, stream: Stream, stream_name: str = None) -> Str
     def _is_should_be_processed(self, stream_name):
         if (
             self.apply_to_streams is not None
+            and self.dont_apply_to_streams is not None
             and stream_name in self.apply_to_streams
             and stream_name in self.dont_apply_to_streams
         ):
             raise ValueError(
                 f"Stream '{stream_name}' can be in either apply_to_streams or dont_apply_to_streams not both."
             )
 
-        return (
-            self.apply_to_streams is None or stream_name in self.apply_to_streams
-        ) and stream_name not in self.dont_apply_to_streams
+        return (self.apply_to_streams is None or stream_name in self.apply_to_streams) and (
+            self.dont_apply_to_streams is None or stream_name not in self.dont_apply_to_streams
+        )
```

Comment:
this part maybe deserves some u-tests (maybe open an issue for that?).

Reply:
There are tests for it.  See  test_stream_refiner.